File: staffapp/src/main/java/com/example/staffapp/TrainerSpecializationCatalog.kt
package com.example.staffapp

/**
 * Справочник специализаций тренера (п.17 репорта).
 * Должен совпадать с StaffOnboardingService::specializationCatalog на бэкенде.
 */
object TrainerSpecializationCatalog {
    const val MAX_SELECTED = 5

    val DEFAULT = listOf(
        "Персональный тренер",
        "Силовые тренировки",
        "Функциональный тренинг",
        "Кроссфит",
        "Йога",
        "Пилатес",
        "Стретчинг",
        "Кардио",
        "Бокс / единоборства",
        "Реабилитация",
        "Похудение",
        "Набор массы",
        "Подготовка к соревнованиям",
        "Детский фитнес",
        "Групповые программы",
    )

    fun parseSelected(raw: String, catalog: List<String> = DEFAULT): List<String> {
        if (raw.isBlank()) return emptyList()
        val lowerCatalog = catalog.map { it.lowercase() }
        return raw.split(',', ';', '|')
            .map { it.trim() }
            .filter { it.isNotEmpty() }
            .mapNotNull { label ->
                val idx = lowerCatalog.indexOf(label.lowercase())
                if (idx >= 0) catalog[idx] else null
            }
            .distinct()
            .take(MAX_SELECTED)
    }

    fun join(selected: List<String>): String = selected.joinToString(", ")
}

File: MatskocaAdmin/MatskocaAdmin/Services/TrainerSpecializationCatalog.swift
import Foundation

/// Справочник специализаций — синхронно с CRM `StaffOnboardingService::specializationCatalog` и Android.
enum TrainerSpecializationCatalog {
    static let maxSelected = 5

    static let `default`: [String] = [
        "Персональный тренер",
        "Силовые тренировки",
        "Функциональный тренинг",
        "Кроссфит",
        "Йога",
        "Пилатес",
        "Стретчинг",
        "Кардио",
        "Бокс / единоборства",
        "Реабилитация",
        "Похудение",
        "Набор массы",
        "Подготовка к соревнованиям",
        "Детский фитнес",
        "Групповые программы",
    ]

    static func parseSelected(_ raw: String, catalog: [String] = Self.default) -> [String] {
        let trimmed = raw.trimmingCharacters(in: .whitespacesAndNewlines)
        guard !trimmed.isEmpty else { return [] }
        let lowerCatalog = catalog.map { $0.lowercased() }
        var out: [String] = []
        for part in trimmed.split(whereSeparator: { ",;|".contains($0) }) {
            let label = part.trimmingCharacters(in: .whitespacesAndNewlines)
            guard !label.isEmpty else { continue }
            if let idx = lowerCatalog.firstIndex(of: label.lowercased()) {
                let canonical = catalog[idx]
                if !out.contains(canonical) {
                    out.append(canonical)
                }
            }
            if out.count >= maxSelected { break }
        }
        return out
    }

    static func join(_ selected: [String]) -> String {
        selected.joined(separator: ", ")
    }
}
